src/components/table/base-table/base-table.js

```diff
@@ -329,51 +339,81 @@ export default class BaseTable extends React.Component {
+                  background: vars.colorNavy,
+                  color: vars.colorWhite,
+                }),
+                columnDefinition.headerClassName,
+                makeClassName({ textAlign: columnDefinition.align || 'left' })
+              )}
+            >
+              {this.headerRenderer(columnDefinition)}
+            </div>
+          )}
+        </ClassNames>
+      );
+    }
     return (
-      <CellMeasurer
-        cache={this.cellMeasurerCache}
-        columnIndex={renderParams.columnIndex}
-        key={renderParams.key}
-        parent={renderParams.parent}
-        rowIndex={renderParams.rowIndex}
-      >
-        {renderParams.rowIndex === 0 ? (
-          <div
-            style={renderParams.style}
-            className={classnames(
-              styles['header-cell'],
-              columnDefinition.headerClassName,
-              styles[`align-${columnDefinition.align || 'left'}`]
-            )}
-          >
-            {this.headerRenderer(columnDefinition)}
-          </div>
-        ) : (
+      <ClassNames>
+        {({ css: makeClassName }) => (
           <div
-            // TODO: move the className and handlers into Cell
-            style={renderParams.style}
             className={classnames(
-              styles[`align-${columnDefinition.align || 'left'}`],
-              {
-                [styles['hovered-row']]:
-                  renderParams.rowIndex === this.state.hoveredRowIndex &&
-                  renderParams.rowIndex !== 0 &&
-                  this.props.onRowClick,
-              },
+              makeClassName({
+                textAlign: columnDefinition.align || 'left',
+                ...(renderParams.rowIndex === this.state.hoveredRowIndex &&
+                renderParams.rowIndex !== 0 &&
+                this.props.onRowClick
+                  ? {
+                      cursor: 'pointer',
+                      transition: vars.transitionStandard,
+                      background: vars.colorGray90,
+                    }
+                  : {}),
+              }),
               columnDefinition.className,
+              columnDefinition.cellClassName,
               columnDefinition.classNameGetter
                 ? columnDefinition.classNameGetter({
                     rowIndex: this.getBodyRowIndex(renderParams.rowIndex),
                     columnKey: columnDefinition.key,
                     height: renderParams.height,
                     width: renderParams.width,
                   })
+                : undefined,
+              columnDefinition.cellClassNameGetter
+                ? columnDefinition.cellClassNameGetter({
+                    rowIndex: this.getBodyRowIndex(renderParams.rowIndex),
+                    columnKey: columnDefinition.key,
+                    height: renderParams.height,
+                    width: renderParams.width,
+                  })
                 : undefined
             )}
+            style={{
```

Comment:
This is passed to a `style` prop. Would it make sense to use object styles and pass it to a `css(columnDef.cellStyleGetter`?

Reply:
I thought about it but honestly I would keep it as is (it was like this before), I don't want to mess up with possible specificity problems 😅